proj/src/main/webapp/chat.js

```diff
@@ -722,23 +722,31 @@ async function addUsernameToMessage(uid, preview) {
 }
 
 // function for when a user adds a picture
-function pfpOnInput() {
+async function pfpOnInput() {
     const input = document.getElementById("pfp-upload");
     const pfp = input.files[0];
 
     const currentUid = firebase.auth().currentUser.uid;
     const pfpStorageRef = firebase.storage().ref(`/profile-pictures/${currentUid}/pfp.png`);
-    pfpStorageRef.put(pfp);
+    const uploadStatus = document.getElementById('upload-status')
+    await pfpStorageRef.put(pfp).then(function() {
+        uploadStatus.innerText = 'upload success';
+        setTimeout(function(){ uploadStatus.innerText = ''; }, 5000);
+    }).catch(function(error) {
+        uploadStatus.innerText = 'upload failed: '+error.message;
+        setTimeout(function(){ uploadStatus.innerText = ''; }, 5000);
+    });
 
-    pfpStorageRef.getDownloadURL().then(function(url) {
+    await pfpStorageRef.getDownloadURL().then(function(url) {
```

Comment:
I think if you really wanted to, you could chain this with the previous await. It might remove the need for async and. awaits.
Something like 
```
pfpStorageRef.put(pfp).then(function() {
      ...
    }).catch(function(error) {
      ...
    }).then(function(){
      return pfpStorageRef.getDownloadURL();
    }).then(function(url){
      ...
    })
```